eZ/Bundle/EzPublishCoreBundle/Converter/RepositoryParamConverter.php

```diff
@@ -0,0 +1,111 @@
+ * @version //autogentag//
+ */
+
+namespace eZ\Bundle\EzPublishCoreBundle\Converter;
+
+use eZ\Publish\API\Repository\ContentService;
+use eZ\Publish\API\Repository\Exceptions\NotFoundException;
+use eZ\Publish\API\Repository\Exceptions\UnauthorizedException;
+use eZ\Publish\API\Repository\LocationService;
+use Sensio\Bundle\FrameworkExtraBundle\Configuration\ParamConverter;
+use Sensio\Bundle\FrameworkExtraBundle\Request\ParamConverter\ParamConverterInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpKernel\Exception\AccessDeniedHttpException;
+use Symfony\Component\HttpKernel\Exception\NotFoundHttpException;
+
+class RepositoryParamConverter implements ParamConverterInterface
+{
+    const LOCATION_ID = 'locationId';
+    const CONTENT_ID = 'contentId';
+
+    const LOCATION_CLASS = 'eZ\Publish\API\Repository\Values\Content\Location';
+    const CONTENT_CLASS = 'eZ\Publish\API\Repository\Values\Content\Content';
+
+    private $supportedClasses;
+
+    /** @var  \eZ\Publish\API\Repository\LocationService */
+    private $locationService;
+
+    /** @var \eZ\Publish\API\Repository\ContentService */
+    private $contentService;
+
+    public function __construct( LocationService $locationService , ContentService $contentService )
+    {
+        $this->locationService = $locationService;
+        $this->contentService = $contentService;
+        $this->supportedClasses = array(
+            self::LOCATION_CLASS,
+            self::CONTENT_CLASS
+        );
+    }
+
+    public function apply( Request $request, ParamConverter $configuration )
+    {
+        try
+        {
+            if ( $configuration->getClass() === self::LOCATION_CLASS )
+            {
+                if ( !$request->attributes->has( self::LOCATION_ID ) )
+                {
+                    return false;
+                }
+
+                $value = $request->attributes->get( self::LOCATION_ID );
+                if ( !$value && $configuration->isOptional() )
+                {
+                    return false;
+                }
+
+                $request->attributes->set( $configuration->getName(), $this->locationService->loadLocation( $value ) );
+                return true;
+            }
+            else if ( $configuration->getClass() === self::CONTENT_CLASS )
+            {
+                if ( !$request->attributes->has( self::CONTENT_ID ) )
+                {
+                    return false;
+                }
+
+                $value = $request->attributes->get( self::CONTENT_ID );
+                if ( !$value && $configuration->isOptional() )
+                {
+                    return false;
+                }
+
+                $request->attributes->set( $configuration->getName(), $this->contentService->loadContent( $value ) );
+                return true;
+            }
+        }
+        catch ( NotFoundException $e )
+        {
+            throw new NotFoundHttpException( 'Requested values not found' );
```

Comment:
HttpExceptions are caught and handled by a Symfony Controller which return a proper response with the right status code rather than a generic 500 error